[limjunxian1] iP

limjunxian1 opened the pull request:
# HelpBuddy
> “Not only should you have a to-do list, but it must become your best friend.” — Jim Kwik ([Source](https://medium.com/@oaklinejournal/25-quotes-from-successful-people-on-the-power-of-writing-to-do-lists-8a0b3729a34c))

HelpBuddy will be your new best buddy. It saves a list of tasks you need to do. It's,
* text-based
* easy to use
* ~~FAST~~ ***SUPER*** FAST

All you need to do is,
1. download it from [here](https://github.com/limjunxian1/ip/releases/download/v0.1/helpbuddy.jar)
2. open your CLI and navigate to the folder where the `helpbuddy.jar` file is
3. run the command `java -jar helpbuddy.jar`
4. add your tasks 
5. allow it to enhance your task organization :grinning:

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Save deadlines
- [ ] Reminders (coming soon)
#
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        new HelpBuddy("data/tasks.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

limjunxian1 committed:
Add HelpBuddy implementation and update project structure

limjunxian1 committed:
Add echo feature

limjunxian1 committed:
Added list and add feature.

limjunxian1 committed:
Add mark as done feature.

limjunxian1 committed:
Add support for tracking three types of task.

limjunxian1 committed:
Add semi-automated testing for HelpBuddy.

limjunxian1 committed:
Add handle errors feature.

limjunxian1 committed:
Add delete feature.

limjunxian1 committed:
Add save feature

limjunxian1 committed:
Merge branch 'branch-level-7'

limjunxian1 committed:
Add feature to understand date and time

limjunxian1 committed:
Merge branch 'branch-level-8'

limjunxian1 committed:
Add more OOP

HelpBuddy class is handling the following actions: intepreting user input, displaying output, storing and loading of data. User input includes multiple action commands as well.

Extracting these behaviors into individual classes allow us to use encapsulation by packaging related behavior into one self-contained unit.

limjunxian1 committed:
Divide classes into packages

Grouping classes that are related for easier identification of classes that are related to one another.

Add a main package helpbuddy with 5 sub-packages, ui, storage, exception, command and task.

limjunxian1 committed:
Merge branch 'add-gradle-support'

limjunxian1 committed:
Add Gradle

limjunxian1 committed:
Add JUnit Tests

limjunxian1 committed:
Create a JAR File

limjunxian1 committed:
Follow the Coding Standard

Changing indentations and order of imports make the flow of code more readable.

limjunxian1 committed:
Add JavaDoc comments

limjunxian1 committed:
Add find feature

limjunxian1 committed:
Merge branch 'branch-A-JavaDoc'

limjunxian1 committed:
Resolve merge conflict

Incoporate master branch and discard branch-A-CodingStandard as Javadoc in master branch is the revised version.

limjunxian1 committed:
Resolve merge conflict

Incorporate master branch and branch-Level-9 changes.

limjunxian1 committed:
Follow coding standard

Remove unnecessary imports in files and correct configurations for checkstyle and test to run smoothly.

Seonlo99 commented on src/main/java/helpbuddy/command/Command.java:
```diff
@@ -0,0 +1,40 @@
+package helpbuddy.command;
```
I think it's really great that you implemented the stretch goal of splitting each of your commands into another class.

Seonlo99 commented on src/test/java/helpbuddy/ui/ParserTest.java:
```diff
@@ -0,0 +1,115 @@
+import helpbuddy.task.Event;
+import helpbuddy.task.ToDo;
+
+public class ParserTest {
```
I think it is good that you have cover a variety of tests. it might be good to also include test for invalid input such as accessing an index that is not on the list.

Seonlo99 commented on src/main/java/helpbuddy/ui/Ui.java:
```diff
@@ -0,0 +1,151 @@
+            String messageOutput = "Here are the tasks in your list:\n";
+            for (int i = 0; i < taskList.getSize(); i++) {
+                int taskIndex = i + 1;
+                messageOutput += "\t" + taskIndex + ". " + taskList.getTask(i) + "\n";
```
I think it might be better if the taskList class could implement Iterable interface so that you can use for each loop instead of using getTask(i).

Seonlo99 commented on src/main/java/helpbuddy/ui/Parser.java:
```diff
@@ -0,0 +1,111 @@
+            if (taskDateDetails.length == 1) {
+                return new AddCommand(new Event(taskName,
+                        LocalDateTime.parse(startTime, formatter),
+                        null
```
I think it is a really good idea to allow the creating of classes and let the respective classes handle the error when the user has missing inputs.

Seonlo99 reviewed (commented):
LGTM.

limjunxian1 committed:
Add GUI to HelpBuddy

limjunxian1 committed:
Merge branch 'branch-level-10'

limjunxian1 committed:
Add Assertions

There are inherent assumptions made in the program.

There is a possibility of violating the inherent assumptions.

Respective assert statements are added to Storage and MainWindow class.

This enables the runtime of program to verify these assertions and surface potential bugs in the code.

limjunxian1 committed:
Improve code quality

Current code structure is not optimal for readability and there are some violations to coding standards.

In the Parser class, part of the logic in parseCommand method is extracted to various private methods for dealing with different types of command that user inputs, reduce the length of the method. Removing unnecessary import statements to reduce clutter in code and obey coding standards.

This allows reader to easily process the code.

limjunxian1 committed:
Add detect duplicate feature

HelpBuddy allows users to add tasks with the same details repeatedly into the task list.

Before adding any task that user inputs, there will be a check to ensure that this task is not present in the task list.

This removes clutter arising from duplicate tasks when users want to see a list of their saved tasks.

limjunxian1 committed:
Merge pull request #1 from limjunxian1/branch-A-Assertions

Add Assertions

limjunxian1 committed:
Merge branch 'master' into branch-A-CodeQuality

limjunxian1 committed:
Merge branch 'master' into C-DetectDuplicates

limjunxian1 committed:
Merge pull request #2 from limjunxian1/branch-A-CodeQuality

Improve code quality

limjunxian1 committed:
Merge branch 'master' of https://github.com/limjunxian1/ip

* 'master' of https://github.com/limjunxian1/ip:
  Improve code quality

limjunxian1 committed:
Merge branch 'master' into C-DetectDuplicates

limjunxian1 committed:
Merge pull request #3 from limjunxian1/C-DetectDuplicates

Add detect duplicate feature

limjunxian1 committed:
Add Ui.png

limjunxian1 committed:
Add User Guide

README contains outdated information on HelpBuddy's implementation.

HelpBuddy's features are documented in README.

This guides users to utilize HelpBuddy's features.

limjunxian1 committed:
Merge pull request #5 from limjunxian1/C-DetectDuplicates

Add Ui.png and User Guide
